incubator/vault/templates/deployment.yaml

```diff
@@ -0,0 +1,52 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "name" . }}
+        release: {{ .Release.Name }}
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        {{- if .Values.vault.dev }}
+        command: ["vault", "server", "-dev"]
+        {{- else }}
+        command: ["vault", "server", "-config", "/vault/config/config.json"]
+        {{- end }}
+        ports:
+        - containerPort: {{ .Values.service.port }}
+        livenessProbe:
```

Comment:
I had trouble getting vault up-and-running with the liveness (and/or readyness probe) enabled, because the time between starting vault and unsealing it was too long.

@jpds Did you test this also with a non-dev config? If so, how did you solve that?

Reply:
Yes, I've used this with before consul and etcd as the backends. Just spun up a new etcd cluster to test it. Didn't have an issue with it starting or unsealing.

Reply:
Stock `incubator/etcd` chart + this as `vault-values.yaml`:

```
vault:
  dev: false
  config:
    storage:
      etcd:
        address: "http://my-release-etcd-0.my-release-etcd.default:2379,my-release-etcd-1.my-release-etcd.default:2379,my-release-etcd-2.my-release-etcd.default:2379
```

Reply:
Thanks for verifying!

Then it seems there was an error at my end. I'm not that familiar with these probes yet...

Reply:
I would take a look at the logs from the pod and see if there's a config error.